godot-macros/src/method_registration/register_method.rs

```diff
@@ -0,0 +1,171 @@
+/*
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at https://mozilla.org/MPL/2.0/.
+*/
+
+use proc_macro2::{Ident, TokenStream as TokenStream2};
+use quote::quote;
+use venial::parse_declaration;
+
+use crate::method_registration::{
+    get_sig, get_signature_info, method_flags, wrap_with_unpacked_params, ReceiverType,
+};
+
+// Convenience function to wrap an object's method into a function pointer
+// that can be passed to the engine when registering a class.
+pub fn gdext_register_method(class_name: &Ident, method_signature: &TokenStream2) -> TokenStream2 {
+    let mut method_signature = parse_declaration(quote! { #method_signature {}})
+        .unwrap()
+        .as_function()
+        .unwrap()
+        .clone();
+
+    method_signature.body = None;
+
+    let signature_info = get_signature_info(&method_signature);
+    let receiver_type = signature_info.receiver_type;
+    let param_idents = &signature_info.param_idents;
+
+    // let class_name = method_def.class_name;
+    let method_name = signature_info.method_name.clone();
+
+    let mut num_args = method_signature.params.len();
+    let (ret_type, has_return_value) = match method_signature.return_ty {
+        None => (quote! {()}, false),
+        Some(ty) => (quote! {#ty}, true),
+    };
+
+    if receiver_type != ReceiverType::Static {
+        num_args -= 1;
+    }
+
+    let wrapped_method = wrap_with_unpacked_params(class_name, &signature_info);
+
+    let method_flags = method_flags(receiver_type);
+
+    let signature = get_sig(ret_type, signature_info.param_types);
+
+    quote! {
+        unsafe {
+            use godot::builtin::meta::*;
+            use godot::builtin::{Variant, StringName};
+            use godot::sys;
+
+            const NUM_ARGS: usize = #num_args;
```

Comment:
You could substitute this directly with `#num_args` wherever it appears, and remove the constant.

`argument_count` further below is still needed though (because it's `u32`).